### PR TITLE
Unarchive project if it's being recreated

### PR DESCRIPTION
**Pull request by srueg**
## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the ./CHANGELOG.md.

### CODE REVIEW

**Comment by Kidswiss** [comment]
LGTM

**Review by Kidswiss — APPROVED** [review]

File: pkg/git/gitlab/gitlab.go [hunk 1]
```diff
@@ -197,6 +197,12 @@ func (g *Gitlab) getProject() error {
 	if err != nil {
 		return err
 	}
+	if project.Archived {
```
Comment (corvus-ch) [inline pkg/git/gitlab/gitlab.go:200]: This is dangerous. A get operation suggests to be read only. Having the unarchive operation within this function, could result in unexpected behavior.

I would prefer to have this function returning an error and the calling code taking care of calling unarchive if it makes sense.

**Review by corvus-ch — CHANGES_REQUESTED** [review]
This change goes down a dangerous route. Please consider to do this differently.

File: pkg/git/gitlab/gitlab.go (hunk 1 above)
Comment (srueg) [reply pkg/git/gitlab/gitlab.go:200]: You're absolutely right. Thinking about it again, I don't think it's a good idea to un-archive a repo. I guess the current behavior of failing if the repo already exists in an archived state is ok.

**Pull request closed without merging**